### PR TITLE
Minor usage refactor for readability and (probably) performance

### PR DESCRIPTION
**Pull request by tech6hutch**
### Description of the PR
Minor usage refactors for readability and (probably) performance
Also fixes a usage min/max bug

### Changes Proposed in this Pull Request (List new items in CHANGELOG.MD)

- Small refactors
- Doc fixes
- Change Possible.resolveLimit to non-static

### Semver Classification
PATCH
- [ ] This PR only includes documentation or non-code changes.
- [x] This PR fixes a bug and does not change the (intended) framework interface.
- [ ] This PR adds methods or properties to the framework interface.
- [ ] This PR removes or renames methods or properties in the framework interface.

### CODE REVIEW

**Comment by tech6hutch** [comment]
Maybe hold off on merging this, for now. I think the usage system could use more commentary, so I I'll add that, as I go through it.

File: src/lib/usage/Tag.js [hunk 1]
```diff
@@ -9,7 +9,7 @@ class Tag {
 	 * @since 0.2.1
 	 * @param {string} members The tag contents to parse
 	 * @param {number} count The position of the tag in the usage string
-	 * @param {boolean} required The type of tag (required or optional)
+	 * @param {number} required The type of tag (0 for optional, anything else for required)
```
Comment (bdistin) [inline src/lib/usage/Tag.js]: 0 optional, 1 semi-required, 2 required

File: typings/index.d.ts [hunk 2]
```diff
@@ -965,11 +965,11 @@ declare module 'klasa' {
 		public max: number;
 		public regex: RegExp;
 
-		private static resolveLimit(limit: string, type: string): number;
+		private resolveLimit(limit: string, limitType: "min" | "max"): number;
```
Comment (tech6hutch) [inline typings/index.d.ts]: Please let me know if this is not a valid TS type

Comment (MrJacz) [reply typings/index.d.ts]: just `string`

**Comment by tech6hutch** [comment]
I see, will fix

**Comment by bdistin** [comment]
No, just no to the last commit

File: src/lib/usage/Possible.js [hunk 3]
```diff
@@ -1,4 +1,7 @@
+// The types towards the beginning require integer min/max limits
 const minMaxTypes = ['str', 'string', 'int', 'integer', 'num', 'number', 'float'];
+// The index of the last type in minMaxTypes that requires integer min/max
+minMaxTypes.intLimitLastIndex = 3;
```
Comment (MrJacz) [inline src/lib/usage/Possible.js]: Shouldn't this just be a variable instead of adding a property to an array?

**Comment by tech6hutch** [comment]
But if it's hardcoded, it'll break if any new types are ever added. I guess a comment to update the number when adding new types would suffice

**Comment by bdistin** [comment]
The Possible class is exactly as it should be in master at this time. This pr should not change how it is from master.

File: src/lib/usage/Possible.js (hunk 3 above)
Comment (tech6hutch) [reply src/lib/usage/Possible.js]: That would also work. I put it on the array since it was related to it.

**Comment by tech6hutch** [comment]
I'm done with this PR now, unless there's anything else.

**Pull request merged**